EIPS/eip-5732.md

```diff
@@ -58,13 +51,19 @@ interface IERC_COMMIT_GENERAL {
         address _from,
         bytes32 _commitment,
         bytes calldata _extraData)
-    payable external returns(uint256 /*timePoint*/);
+    payable external returns(uint256 timePoint);
 }
 ```
 
+1. A compliant contract MUST implement the `IERC_COMMIT_CORE` interface.
+2. A compliant contract SHOULD implement the `IERC_COMMIT_GENERAL` interface.
+3. A compliant contract that implements the `IERC_COMMIT_GENERAL` interface MUST accept `commit(_commitment)` as equivalent to `commitFrom(msg.sender, _commitment, [/*empty array*/])`.
+4. The `timePoint` return value of `commitFrom` is RECOMMENDED to use `block.timestamp` or `block.number` or a number that indicates the ordering of different commitments. When `commitFrom` is being called.
+5. A compliant contract that implements `IERC_COMMIT_GENERAL` MUST emit event `Commit` when a commitment is accepted and recorded. In the parameter of both `Commit` and the `commitFrom` method, the `_timePoint` is a time-point-representing value that represents ordering of commitments in which a latter commitment will always have a greater ?????? value than a former commitment, such as `block.timestamp` or `block.number` or other time scale chosen by implementing contracts. The `extraData` is reserved for future behavior extension. If the `_from` is different from the TX signer, it is RECOMMENDED that compliant contract SHOULD validate signature for `_from`. For EOAs this will be validating its ECDSA signatures on chain. For smart contract accounts, it is RECOMMENDED to use [EIP-1271](./eip-1271.md) to validate the signatures.
+
 4. One or more methods of a compliant contract MAY be used for reveal.
 
-But there MUST be a way to supply an extra field of `secret_salt`, so that committer can later open the `secret_salt` in the reveal TX that exposes the `secret_salt`. The size and location of `secret_salt` is intentionally unspecified in this EIP to maximize flexibility for integration.
+   But there MUST be a way to supply an extra field of `secret_salt`, so that committer can later open the `secret_salt` in the reveal TX that exposes the `secret_salt`. The size and location of `secret_salt` is intentionally unspecified in this EIP to maximize flexibility for integration.
```

Comment:
```suggestion
But there MUST be a way to supply an extra field of `secret_salt`, so that committer can later open the `secret_salt` in the reveal TX that exposes the `secret_salt`. The size and location of `secret_salt` is intentionally unspecified in this EIP to maximize flexibility for integration.
```